diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 8dc3e85..b27e053 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -2197,4 +2197,15 @@ my %targets = (
         cflags           => add("/POINTER_SIZE=64=ARGV"),
         pointer_size     => "64",
     }
+,
+### WASM targets
+    "wasm32-wasi" => {
+        inherit_from     => ["BASE_unix"],
+        cflags           => add("--target=wasm32-wasi"),
+        cxxflags         => add("--target=wasm32-wasi"),
+        lib_cppflags     => add("-DL_ENDIAN"),
+        bn_ops           => "THIRTY_TWO_BIT",
+        pointer_size     => "32",
+    },
+
 );
